pkg/io/data.go

```diff
@@ -155,6 +157,31 @@ type jsonCounterEvent struct {
 	Values map[string]float64 `json:"args,omitempty"`
 }
 
+type tempJsonCounterEvent struct {
+	jsonEventCore
+	Values map[string]string `json:"args,omitempty"`
+}
+
+func (ce *jsonCounterEvent) UnmarshalJSON(data []byte) error {
+	t := &tempJsonCounterEvent{}
+	if err := json.Unmarshal(data, t); err != nil {
+		return err
+	}
+
+	ce.jsonEventCore = t.jsonEventCore
+	ce.Values = make(map[string]float64)
+	for k, v := range t.Values {
+		f, err := strconv.ParseFloat(v, 64)
+		if err != nil {
+			return err
+		}
+
+		ce.Values[k] = f
+	}
+
+	return nil
+}
```

Comment:
I believe this code means that it can no longer decode counter events that contain numbers not represented as strings.

The specification for counter events does not specify the type of the values, but its examples only show raw numbers, not strings: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.msg3086636uq

I'm not currently at a computer I can test the following code on, so this may have errors in, but perhaps something like this would be better, what is your opinion?

```suggestion
	Values map[string]float64 `json:"args,omitempty"`
}

type numberOrString struct {
	number float64
	str         string
}

func (nos *numberOrString) UnmarshalJSON(data []byte) error {
	if err := json.Unmarshal(data, &nos.str); err == nil {
		return nil
	}
	
	return json.Unmarshal(data, &nos.number);
}

type tempJsonCounterEvent struct {
	jsonEventCore
	Values map[string]numberOrString `json:"args,omitempty"`
}

func (ce *jsonCounterEvent) UnmarshalJSON(data []byte) error {
	t := &tempJsonCounterEvent{}
	if err := json.Unmarshal(data, t); err != nil {
		return err
	}

	ce.jsonEventCore = t.jsonEventCore
	ce.Values = make(map[string]float64)
	for k, numberOrStr := range t.Values {
		value := numberOrStr.number
		
		if numberOrStr.str != "" {
			f, err := strconv.ParseFloat(v, 64)
			if err != nil {
				return err
			}
			value = f
		}
		
		ce.Values[k] = value
	}

	return nil
}
```

Reply:
Direction LGTM.

There are a few small syntax error so I couldnt apply it as-is, but I have made my own edit accordingly and pushed an update